src/sentry/search/events/builder/metrics.py

```diff
@@ -760,6 +778,29 @@ def _environment_filter_converter(self, search_filter: SearchFilter) -> WhereTyp
         else:
             return env_conditions[0]
 
+    def _evaluate_metric_matches_function(
+        self, metric_id: int | None, expected_prefix: set[str] | str
+    ):
+        if type(expected_prefix) is str:
+            expected_prefix = {expected_prefix}
+
+        if metric_id is None or metric_id == -1:
+            return True
```

Comment:
The `MetricQueryBuilderTest.test_multiple_references_only_resolve_index_once` test was providing `-1` as the `metric_id` for `measurements.lcp` and it would fail to resolve in the StringIndexer. Is this some sort of test artefact or does `-1` mean something for metric IDs, similar to how -1 means all projects in project selection?

Reply:
and I think if we do [this](https://github.com/getsentry/sentry/pull/69733/files#r1583644084), we don't have to handle that test case explicitly

Reply:
Ah, I realized that the test was mocking the `-1` value so there isn't a code path that sets `-1` as the value for a metric ID. I'm going to get rid of this and mock the `reverse_resolve` for now. The test doesn't update the string indexer so the reverse resolve fails